added application layers

IvanLinnik opened the pull request:
move controller, item, service, parser classes
added lombok constructors and standard methods generation
refactor parser class

IvanLinnik committed:
added application layers
move controller, item, service, parser classes
added lombok constructors and standard methods generation
refactor parser class

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/controllers/ItemController.java:
```diff
@@ -10,17 +11,13 @@
 
-
-    @Autowired
     public ItemController(ItemService itemService) {
```
replace with lombok

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/dto/Item.java:
```diff
@@ -0,0 +1,15 @@
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Item {
```
rename it to RssItem

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/services/ItemService.java:
```diff
@@ -1,24 +1,21 @@
 import java.util.List;
 
 @Service
 public class ItemService {
```
rename to LostFilmService and add interface

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/services/ItemService.java:
```diff
@@ -1,24 +1,21 @@
-        List<Item> list = new ArrayList<>();
         try {
-            list = LostFilmParser.getItemList();
+            return new LostFilmParser().getItemList();
```
LostFilmParser is a Component so inject it accordingly

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/services/LostFilmParser.java:
```diff
@@ -0,0 +1,63 @@
+
+@Component
+public class LostFilmParser {
+    //    @Value("${uri}")
```
why you commented this?

IvanLinnik replied:
А если я хочу хочу поставить аннотацию @Value, то поле не должно быть ни static ни final?

mmarusich commented on src/main/resources/application.properties:
```diff
@@ -1 +1,2 @@
-
+uri=https://www.lostfilm.tv/rss.xml
```
uri is to general, rename to `lostfilm.rss.url`

mmarusich commented on src/main/resources/application.properties:
```diff
@@ -1 +1,2 @@
-
+uri=https://www.lostfilm.tv/rss.xml
+mainTag=item
```
`lostfilm.rss.tag`

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/services/LostFilmParser.java:
```diff
@@ -0,0 +1,63 @@
+    //    @Value("${mainTag}")
+    private static final String MAIN_TAG = "item";
+
+    LostFilmParser() {
```
this constructor is redundant

IvanLinnik replied:
но если оставить контструтор по умолчанию, то можно будет создать экземпляр класса из другого пакета

IvanLinnik committed:
repair all according to review comments
add SpringConfig.java
move variables into application.properties
add lombok.config (if i delete lombok.config file it works anyway with auto generated constructor)

IvanLinnik commented:
может уже и без lombok.config работает? или где то могут кешироваться настройки?

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/SpringConfig.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.context.annotation.PropertySource;
+
+@Configuration
+@PropertySource("classpath:application.properties")
```
this is redundant, application.properties is reserved by spring boot and will be processed automatically

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/services/LostFilmServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+        try {
+            return parser.getItemList();
+        } catch (ParserConfigurationException | IOException | SAXException e) {
+            log.error("", e);
```
provide some error message text

mmarusich reviewed (approved):
few small comments
